features/steps/test_behave.py

```diff
@@ -1,108 +1,215 @@
-      map(_right_decorator, list(sb - sa)))
+    """ Computes both left/right diff between lists `a` and `b` """
+    sa, sb = set(a), set(b)
+    return (map(_left_decorator, list(sa - sb)),
+        map(_right_decorator, list(sb - sa)))
+
+
+def package_version_lists(pkg, list_ver):
+    """ Select package versions """
+    new_list = [x for x in list_ver if re.search('^' + pkg, x)]
+    assert len(new_list) == 1
+    return str(new_list[0])
+
+
+def package_absence(pkg, list_ver):
+    """ Select package versions """
+    new_list = [x for x in list_ver if re.search('^' + pkg, x)]
+    assert len(new_list) == 0
+    return None
+
 
 def execute_dnf_command(cmd, reponame):
-  " Execute DNF command with default flags and the specified `reponame` enabled "
-  flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
-  return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+    """ Execute DNF command with default flags and the specified `reponame` enabled """
+    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
+    return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+
+
+def execute_dnf_command_notinstall(cmd, reponame):
+    """ Execute DNF command with default flags and the specified `reponame` enabled """
+    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
+    return subprocess.call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+
 
 def execute_rpm_command(pkg, action):
-  " Execute given action over specified pkg(s) "
-  if not isinstance(pkg, list):
-    pkg = [pkg]
-  if action == "remove":
-    action = RPM_ERASE_FLAGS
-  elif action == "install":
-    action = RPM_INSTALL_FLAGS
-    pkg = decorate_rpm_packages(pkg)
-  return subprocess.check_call(['rpm'] + action + pkg, stdout=subprocess.PIPE)
+    """ Execute given action over specified pkg(s) """
+    if not isinstance(pkg, list):
+        pkg = [pkg]
+    if action == "remove":
+        action = RPM_ERASE_FLAGS
+    elif action == "install":
+        action = RPM_INSTALL_FLAGS
+        pkg = decorate_rpm_packages(pkg)
+    return subprocess.check_call(['rpm'] + action + pkg, stdout=subprocess.PIPE)
+
 
 def piecewise_compare(a, b):
-  " Check if the two sequences are identical regardless of ordering "
-  return sorted(a) == sorted(b)
+    """ Check if the two sequences are identical regardless of ordering """
+    return sorted(a) == sorted(b)
+
 
 def split(pkg):
-  return [p.strip() for p in pkg.split(',')]
+    return [p.strip() for p in pkg.split(',')]
+
 
 @given('I use the repository "{repo}"')
 def given_repo_condition(context, repo):
-  " :type context: behave.runner.Context "
-  assert repo
-  assert os.path.exists('/build/' + repo)
-  a = [os.remove(p) for p in os.listdir('/repo')]
-  subprocess.check_call(['cp -rs /build/' + repo + '/* /repo/'], shell=True)
+    """ :type context: behave.runner.Context """
+    assert repo
+    context.repo = repo
+    assert context.repo
```

Comment:
In this case the check is redundant because `repo` is a regular variable of the `context` object.
Checking like this would make sense if `repo` was a `property`, that is, there was some logic executed each time `set` or `get` is called:

```
@property
def value(self):
  return self.value

@property.setter
def value(self, value):
  if ....:
    self.value = value
    
```